File: BI-TJV/semestral/rest-client/app/src/main/java/cz/wrzecond/restsys/employee/EmployeeCreateListener.java
package cz.wrzecond.restsys.employee;

import android.app.Dialog;
import android.content.DialogInterface;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.widget.SwitchCompat;

import cz.wrzecond.restsys.R;
import cz.wrzecond.restsys.CreateListener;
import wrzecond.employee.TjvEmployeeCreateDTO;
import wrzecond.employee.TjvEmployeeReadDTO;

public class EmployeeCreateListener extends CreateListener<TjvEmployeeReadDTO> {

    private Boolean adminValue = null;

    public EmployeeCreateListener (EmployeeActivity activity, Dialog dialog, TjvEmployeeReadDTO dto) { super(activity, dialog, dto); }

    @Override
    public void onShow (DialogInterface di) {
        // Init dialog
        EditText username = dialog.findViewById(R.id.create_employee_username);
        EditText password = dialog.findViewById(R.id.create_employee_password);

        EditText firstName = dialog.findViewById(R.id.create_employee_first_name);
        EditText lastName = dialog.findViewById(R.id.create_employee_last_name);
        SwitchCompat admin = dialog.findViewById(R.id.create_employee_admin);
        admin.setOnCheckedChangeListener( (v, checked) -> adminValue = checked );

        if ( dto != null ) {
            username.setText(dto.getUsername());
            firstName.setText(dto.getFirstName());
            lastName.setText(dto.getLastName());
            admin.setChecked(dto.getAdmin());

            Button button = dialog.findViewById(R.id.create_button);
            button.setText(R.string.button_edit);
        }
        else
            adminValue = false; // pre-fill

        // Perform
        dialog.findViewById(R.id.create_button).setOnClickListener( view -> {
            // Create DTO
            TjvEmployeeCreateDTO dto = new TjvEmployeeCreateDTO(
                username.getText().toString(),
                password.getText().toString().isEmpty() ? null : password.getText().toString(),

                firstName.getText().toString(),
                lastName.getText().toString(),
                adminValue
            );
            scheduleRequest(dto, "/employees/");
        });
    }

}

File: BI-KOT/semestral/rest-client/app/src/main/java/cz/wrzecond/restsys/employee/EmployeeCreateListener.kt
package cz.wrzecond.restsys.employee

import android.app.Dialog
import android.content.DialogInterface
import android.view.View
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.widget.SwitchCompat
import cz.wrzecond.restsys.CreateListener
import cz.wrzecond.restsys.R
import wrzecond.TjvEmployeeReadDTO
import wrzecond.TjvEmployeeUpdateDTO

class EmployeeCreateListener (activity: EmployeeActivity, dialog: Dialog, dto: TjvEmployeeReadDTO?)
: CreateListener<TjvEmployeeReadDTO> (activity, dialog, dto) {

    private var adminValue: Boolean? = null

    override fun onShow (di: DialogInterface) {
        // Init dialog
        val username = dialog.findViewById<EditText>(R.id.create_employee_username)
        val password = dialog.findViewById<EditText>(R.id.create_employee_password)

        val firstName = dialog.findViewById<EditText>(R.id.create_employee_first_name)
        val lastName = dialog.findViewById<EditText>(R.id.create_employee_last_name)
        val admin: SwitchCompat = dialog.findViewById(R.id.create_employee_admin)
        admin.setOnCheckedChangeListener { _, checked -> adminValue = checked }

        if (dto != null) {
            username.setText(dto.username)
            firstName.setText(dto.firstName)
            lastName.setText(dto.lastName)
            admin.isChecked = dto.admin
            val button = dialog.findViewById<Button>(R.id.create_button)
            button.setText(R.string.button_edit)
        }
        else
            adminValue = false // pre-fill

        // Perform
        dialog.findViewById<View>(R.id.create_button).setOnClickListener {
            // Create DTO
            val dto = TjvEmployeeUpdateDTO (
                username.text.toString(),
                if (password.text.toString().isEmpty()) null else password.text.toString(),
                firstName.text.toString(),
                lastName.text.toString(),
                adminValue
            )
            scheduleRequest(dto, "/employees/")
        }
    }

}